src/main/java/com/graczdev/ipcalculator/calculator/IPAddress.java

```diff
@@ -0,0 +1,99 @@
+package com.graczdev.ipcalculator.calculator;
+
+import panda.std.Option;
+import panda.std.Quad;
+import panda.std.stream.PandaStream;
+import panda.utilities.StringUtils;
+import panda.utilities.text.Joiner;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+
+public class IPAddress {
+
+    private final InetAddress address;
+    private final String original;
+    private final AddressParts decimalParts;
+    private final AddressParts binaryParts;
+
+    public IPAddress(String address) {
+        this.address = Option.attempt(UnknownHostException.class, () -> InetAddress.getByName(address))
+                .orThrow(() -> new RuntimeException(address + " isn't IP address!"));
```

Comment:
Jak to działa?

Reply:
`Option.attempt(UnknownHostException.class, () -> InetAddress.getByName(address))`
robi optiona z `InetAddress.getByName(address)` jeśli wywali wyjątek `UnknownHostException` to będzie empty option

`.orThrow(() -> new RuntimeException(address + " isn't IP address!"));` pobiera zawartość z optiona, jeśli będzie empty to wyrzuca wyjątek `new RuntimeException(address + " isn't IP address!")`